### PR TITLE
setup for resizable windows, drop weight

### PR DESCRIPTION
**Pull request by taiyu-len**
as it says, it drops the weight method for calculating size,
and instead uses previous width/height.

### CODE REVIEW

File: sway/layout.c [hunk 1]
```diff
@@ -186,40 +186,62 @@ void arrange_windows(swayc_t *container, int width, int height) {
...
-		total_weight += child->weight;
-	}
-
+	x = y = 0;
+	double scale = 0;
 	switch (container->layout) {
 	case L_HORIZ:
 	default:
-		sway_log(L_DEBUG, "Arranging %p horizontally", container);
+		//Calculate total width
```
Comment (SirCmpwn) [inline sway/layout.c]: Minor style - space at the start of comments

**Comment by SirCmpwn** [comment]
How does this behave, in plain english?

**Comment by taiyu-len** [comment]
it sums up the total width of child containers,
(if a child container has width/height 0 it is given a default value of to fit in container nicely)
it then sets the scaling to be `parent->width / total_width`
if it is unchanged, scaling is 1, and widths/height are unchanged.
if total width is smaller (as a result of removing / resizing container), the scaling is 1.n, and the widths/height of all the children are increased to fill the parent container.
if total width is larger (as a result in inserting / resizing container), the scaling is 0.n, and the widths/height of all the children are decreased to fill the parent container.

**Comment by SirCmpwn** [comment]
Looks good from my testing.
